src/main/java/com/o19s/es/ltr/rest/RestCreateModelFromSet.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright [2017] Wikimedia Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.o19s.es.ltr.rest;
+
+import com.o19s.es.ltr.action.CreateModelFromSetAction;
+import com.o19s.es.ltr.action.CreateModelFromSetAction.CreateModelFromSetRequestBuilder;
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.ParsingException;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.RestStatusToXContentListener;
+
+import java.io.IOException;
+
+public class RestCreateModelFromSet extends FeatureStoreBaseRestHandler {
+    private static final ObjectParser<ParserState, Void> PARSER;
+    static {
+        PARSER = new ObjectParser<>("model", ParserState::new);
+        PARSER.declareString(ParserState::setName, new ParseField("name"));
+        PARSER.declareString(ParserState::setModelType, new ParseField("type"));
+        PARSER.declareString(ParserState::setModelDefinition, new ParseField("definition"));
```

Comment:
makes sense, not sure how to do that cleanly yet.
I'll try to change the REST api for this and use the Content-Type http header to read the model name.
Doing this while parsing is bit tricky since I'd need to change the model representation from String to either String/XContent ...